src/models/base/FirestoreCollection.ts

```diff
@@ -20,14 +21,21 @@ export default abstract class FirestoreCollection<T> {
 
   /** Returns a wrapped query promise that converts the data. */
   protected static list<E>(
-    query: firestore.Query<E>
-  ): Promise<FirestoreModel<E>[]> {
-    return query
-      .get()
-      .then((querySnapshot: firestore.QuerySnapshot<E>) =>
-        querySnapshot.docs.map(
+    query: firestore.Query<E>,
+    lastDocument?: firestore.QueryDocumentSnapshot<E>
+  ): Promise<ScholarshipCollectionList<E>> {
+    let firestoreQuery: firestore.Query<E> = query.limit(5);
+    if (lastDocument !== undefined)
+      firestoreQuery = firestoreQuery.startAfter(lastDocument);
+
+    return firestoreQuery.get().then((qSnap: firestore.QuerySnapshot<E>) => {
+      return {
+        next: () => this.list(query, qSnap.docs[qSnap.docs.length - 1]),
+        results: qSnap.docs.map(
           (doc) => new FirestoreModel<E>(doc.ref, doc.data())
-        )
-      );
+        ),
+        empty: qSnap.empty,
```

Comment:
You should try conditionally setting `next` (or instead of `empty`, `hasNext`) based on the amount of results you get vs. requested. E.g. you could do:

```ts
        next: () => this.list(query, qSnap.docs[qSnap.docs.length - 1]),
        results: qSnap.docs.map(
          (doc) => new FirestoreModel<E>(doc.ref, doc.data())
        ),
        hasNext: qSnap.empty || qSnap.size < limit,
```

or
 
```ts
        next: () => qSnap.empty ? null : this.list(query, qSnap.docs[qSnap.docs.length - 1]),
        results: qSnap.docs.map(
          (doc) => new FirestoreModel<E>(doc.ref, doc.data())
        ),
        empty: qSnap.empty,
```

I think the first version probably looks nicer so you don't have to deal with null `next` values.

Reply:
Nice I went with the first option.